packages/server-admin-ui/src/views/DataBrowser.js

```diff
@@ -168,35 +206,150 @@ class DataBrowser extends Component {
       this.props.webSocket.messageHandler = null
     }
   }
-  
+
   componentDidMount() {
     this.fetchSources()
     this.subscribeToDataIfNeeded()
+    this.fetchBaseDeltas()
+    this.fetchUnitList()
   }
 
   componentDidUpdate() {
     this.subscribeToDataIfNeeded()
   }
-  
+
   componentWillUnmount () {
     this.unsubscribeToData()
   }
 
   handleContextChange(event) {
     this.setState({...this.state, context: event.target.value})
-    localStorage.setItem(contextStorageKey, event.target.value)  
+    localStorage.setItem(contextStorageKey, event.target.value)
   }
 
   handleSearch(event) {
     this.setState({...this.state, search: event.target.value})
-    localStorage.setItem(searchStorageKey, event.target.value)  
+    localStorage.setItem(searchStorageKey, event.target.value)
   }
 
   handleMeta(event) {
     this.setState({...this.state, includeMeta: event.target.checked})
     localStorage.setItem(metaStorageKey, event.target.checked)
   }
 
+  handleEdit(event) {
+    this.setState({...this.state, edit: event.target.checked})
+    localStorage.setItem(editStorageKey, event.target.checked)
+  }
+
+  fetchUnit(basedeltas){
+    basedeltas.map((resArray) => { //context also?
+      resArray.updates.map((update) => {
+        if (update.meta){
+          update.meta.map((metas, index) => {
+            if (metas.path && metas.value.units) {
+              let units = this.state.unit
+              units[metas.path] = metas.value.units
+              this.setState({...this.state, unit: units})
```

Comment:
The code above works, but not why you think and it is against React guidelines.
Check this code:
```
x = {a: 1}
y = x
y.a = 2
console.log(x)
//prints { a: 2 }
```
and read
https://stackoverflow.com/questions/37755997/why-cant-i-directly-modify-a-components-state-really/40309023#40309023 and one way to fix it.